facia-tool/public/js/utils/drag-dispatcher.js

```diff
@@ -56,16 +56,22 @@ function handleMedia ({sourceItem, mediaItem}, targetItem, targetGroup) {
 
 function handleInternalClass ({sourceItem, sourceGroup}, targetItem, targetGroup) {
 
-    if (targetItem instanceof Article && targetItem.id() === sourceItem.id) {
-        return;
-    }
-
     var {position, target, isAfter} = normalizeTarget(sourceItem, targetItem, targetGroup);
 
-    removeById(targetGroup.items, urlAbsPath(sourceItem.id));
+    var insertAt = 0;
+
+    for (var i = 0; i < targetGroup.items().length; i++) {
```

Comment:
I'd cache `targetGroup.items()` outside the for loop to avoid calling that function twice at each iteration